backend/application/api/src/main/kotlin/io/raemian/api/task/service/TaskService.kt

```diff
@@ -60,6 +62,12 @@ class TaskService(
         taskRepository.delete(task)
     }
 
+    @Transactional(readOnly = true)
+    fun findGoalTaskCountMap(goalIds: List<Long>): Map<Long, Int> {
+        val goalTaskCounts = taskJdbcQueryRepository.findAllGoalTaskCountByGoalIdIn(goalIds)
+        return goalTaskCounts.associate { it.goalId to it.taskCount }
```

Comment:
```suggestion
        val goalCountMap = goalTaskCounts
                .associate { it.goalId to it.taskCount }
                .map { GoalCountMap(goalId = it.key, taskCount = it.value)}
        return goalCountMap
```

Map을 반환하는것도 좋지만 
data class  하나 만들어서 반환하는게 가독성이 더 좋을지도...?

Reply:
이 코멘트를 남긴 이유가 Key, Value값의 의미가 모호해서라 생각하는데.. 그 말에 전적으로 동의...

그런데 위 코드에서는 type을 명시하지 않아서 알 수 없었지만 goalTaskCounts는 List<GoalTaskCountQueryResult>이고 GoalTaskCountQueryResult는 필드로 goalId, taskId를 가지고 있는 상황..

효율성 때문에 Map을 반환했는데 가독성 측면에서 List<GoalTaskCountQueryResult> 그대로 반환하는 게 더 좋을까?

Reply:
리스트를 반환하는 함수 하나를 만들고

만든 함수를 map 으로 가공하는 함수를 하나 더 만들어서 사용하는건 어떨까?

Reply:
map이 json으로 어떻게 만들어지는지 기억이 안 나는데, 조금 특이했던 것으로 기억합니다.
저는 json으로 봤을 때 이쁜 방법이면 좋을 것 같습니다 ㅎㅎ

Reply:
이 부분만 보면 Controller에서 사용한다고 생각될 것 같네요..ㅎㅎ

우선 이 메서드의 반환 값은 Response로 들어가는 게 아니라 GoalQueryService에서 사용하기 위해 만든 거라 map이 json형태로 어떻게 변환되는지는 지금은 고려하지 않아도 될 것 같습니다..!

추가로 메서드 분리하는 방향 좋은 것 같습니다~!